spatialmath/sphere.go

```diff
@@ -14,68 +14,34 @@ import (
 // sphere density corresponding to the total amount of points.
 const defaultTotalSpherePoints = 100
 
-// SphereCreator implements the GeometryCreator interface for sphere structs.
-type sphereCreator struct {
-	radius float64
-	pointCreator
-	label string
-}
-
 // sphere is a collision geometry that represents a sphere, it has a pose and a radius that fully define it.
 type sphere struct {
 	pose   Pose
 	radius float64
 	label  string
 }
 
-// NewSphereCreator instantiates a SphereCreator class, which allows instantiating spheres given only a pose which is applied
-// at the specified offset from the pose. These spheres have a radius specified by the radius argument.
-func NewSphereCreator(radius float64, offset Pose, label string) (GeometryCreator, error) {
+// NewSphere instantiates a new sphere Geometry.
+func NewSphere(radius float64, offset Pose, label string) (Geometry, error) {
```

Comment:
Suggest swapping offset and radius ordering so that Boxes and Spheres both have their Pose arguments first.

Reply:
Good catch, definitely should do that.